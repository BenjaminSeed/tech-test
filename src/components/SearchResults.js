import React from "react";
import PropTypes from "prop-types";
import "../styles/search-results.css";

const SearchResults = ({ results }) => {
  if (!results.length) {
    return <p>No Results</p>;
  } else {
    return (
      <div className="search-results">
        {results.map((image) => (
          <img
            key={image} // check this
            className="card-image"
            src={image}
            alt="spaceImage"
          />
        ))}
      </div>
    );
  }
};

export default SearchResults;

SearchResults.propTypes = {
  results: PropTypes.array.isRequired,
};
