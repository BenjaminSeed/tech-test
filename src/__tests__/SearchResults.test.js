import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("Search Results", () => {
  const results = [
    {
      image: "https://images-assets.nasa.gov/image/PIA07767/PIA07767~thumb.jpg",
    },
  ];
  it("renders an image when search function used", () => {
    render(<SearchResults results={results} />);
    expect(results).toEqual(results);
  });
});

describe("Search Results-snapshot", () => {
  it("renders Search result without an actual result", () => {
    const { asFragment } = render(<SearchResults results={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
