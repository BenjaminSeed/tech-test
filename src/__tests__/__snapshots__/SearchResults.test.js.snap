// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results-snapshot renders Search result without an actual result 1`] = `
<DocumentFragment>
  <p>
    No Results
  </p>
</DocumentFragment>
`;
